spec/consensus/light-client/detection.md

```diff
@@ -1,3 +1,68 @@
-# Detection
+# Failure Detection
 
-TODO
+The light client must be able to detect failures in the trust assumptions.
+Ideally, it should never trust a header, which was forged by malicious
+validators.
+
+## Connectivity requirement
+
+To meet the above requirement, it must be connected to at least one honest
+full node. This is sufficient so long as there's no fork on the main chain.
+
+If there is a fork on the main chain, it means that two full nodes have decided
+on two different headers for the same height. Then the requirement would at
+least require that the lite client is **connected to one honest full node on
+each branch of the fork**.
+
+_Remark_: +1/3 of malicious validators can create an unlimited number of forks.
+
+However, if we assume that correct full node is going to halt in case of a fork
```

Comment:
this does not change anything for the light client because it still needs to be connected to the "right" set of nodes (i.e. on both branches) in order to detect a fork